src/main/java/duke/DukeStorage.java

```diff
@@ -0,0 +1,165 @@
+                throw new DukeException("Invalid /by field. Removing...");
+            }
+            try {
+                LocalDateTime dateTime = LocalDateTime.parse(currLine[2].substring(1), format);
+                return new Deadline(description, dateTime);
+            } catch (DateTimeParseException e) {
+                throw new DukeDateFormatException();
+            }
+            // Fallthrough
+        }
+        case("Event"): {
+            if (currLine[2].length() < 1) {
+                throw new DukeException("Invalid /at field. Removing...");
+            }
+            try {
+                LocalDateTime at = LocalDateTime.parse(currLine[2].substring(1), format);
+                return new Event(description, at);
+            } catch (DateTimeParseException e) {
+                throw new DukeDateFormatException();
+            }
+            // Fallthrough
+        }
+        default :
+            throw new DukeException("Invalid task. Removing...");
+            // Fallthrough
+        }
+    }
+
+    /**
+     * Load tasks from stored history file.
+     *
+     * @return the array list
+     * @throws IOException   the io exception
+     * @throws DukeException the duke exception
+     */
+    protected ArrayList<Task> load() throws IOException, DukeException {
+        File file = new File(fileName);
+        try {
+            if (!file.exists()) {
+                if (file.createNewFile()) {
+                    System.out.println("New history created!");
+                }
+            }
+        } catch (IOException ex) {
+            System.out.println(ex.getMessage());
+        }
+        Scanner sc = new Scanner(file);
+        ArrayList<Task> taskList = new ArrayList<>();
+        while (sc.hasNext()) {
+            taskList.add(parse(sc.nextLine()));
+        }
+        return taskList;
+    }
+
+    /**
+     * Unload tasks into stored history file.
+     *
+     * @param taskList the task list
+     * @throws IOException the io exception
+     */
+    protected void unload(DukeTaskList taskList) throws IOException {
+        StringBuilder content = new StringBuilder();
+        for (Task t : taskList.getTaskList()) {
+            String currTask = "";
+            switch (t.getClass().getSimpleName()) {
+            case "Todo" : {
+                currTask = "Todo | " + t.getDescription();
+                break;
+            }
+            case "Deadline": {
+                currTask = "Deadline | " + t.getDescription() + " | "
+                        + ((Deadline) t).getSimpleBy();
+                break;
+            }
+            case "Event": {
+                currTask = "Event | " + t.getDescription() + " | "
+                        + ((Event) t).getSimpleAt();
+                break;
+            }
+            }
+            content.append(currTask);
+            content.append("\n");
+        }
+        FileWriter fw = new FileWriter(this.fileName);
```

Comment:
Personally I don't feel that fw is a very clear acronym. Maybe a clearer variable name here?